### PR TITLE
Making the SideTabs aware of errors in the fields

### PR DESCRIPTION
**Pull request by wavesoft**
This pull request makes `SideTabs` aware of errors in it's fields.

image

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @MatApple

**Comment by daiyiwhale** [comment]
This is fantastic! ✨ 

I remember hearing rumors from design at some point that they wanted errors to be represented by little red badges on the side tabs, might want to confirm with @leemunroe?

**Comment by leemunroe** [comment]
Oh nice! ✨

No badges required. This is inline with [what we had designed](https://invis.io/R381VNZXA).

File: src/js/components/SideTabs.js [hunk 1]
```diff
@@ -26,9 +26,17 @@ class SideTabs extends React.Component {
     let {selectedTab, tabs} = this.props;
 
     return tabs.map((tab, index) => {
-      let {title, selectValue} = tab;
+      let {title, selectValue, definition} = tab;
+
+      // Check if at least one field has errors
+      let hasErrors = definition && definition.reduce(function ( lastErrors, def ) {
```
Comment (jfurrow) [inline src/js/components/SideTabs.js]: #nit Let's kill the whitespace around the parameters of this funciton.

**Comment by jfurrow** [comment]
Awesome!

File: src/js/components/SideTabs.js (hunk 1 above)
Comment (wavesoft) [inline src/js/components/SideTabs.js]: Done!

**Comment by wavesoft** [comment]
@mesosphere-ci retest this please

File: src/js/components/SideTabs.js [hunk 2]
```diff
@@ -26,9 +26,17 @@ class SideTabs extends React.Component {
     let {selectedTab, tabs} = this.props;
 
     return tabs.map((tab, index) => {
-      let {title, selectValue} = tab;
+      let {title, selectValue, definition} = tab;
+
+      // Check if at least one field has errors
+      let hasErrors = definition && definition.reduce(function (lastErrors, def) {
+        return lastErrors || !!def.showError;
```
Comment (rcorral) [inline src/js/components/SideTabs.js]: We typically try to avoid abbreviations, could we rename `def` to `field` or `fieldDefinition`?

**Comment by rcorral** [comment]
This is really great :)

File: src/js/components/SideTabs.js (hunk 2 above)
Comment (wavesoft) [inline src/js/components/SideTabs.js]: `field` is a good alternative. Done.

**Comment by rcorral** [comment]
Awesome work!

**Comment by daiyiwhale** [comment]
Rad!

**Pull request merged**